nat-lab/tests/test_dns.py

```diff
@@ -15,23 +15,70 @@
     new_connection_by_tag,
 )
 from utils.process import ProcessExecError
+from utils.router import IPStack
 
 DNS_IPV4_SERVER_ADDRESS = config.LIBTELIO_DNS_IPV4
 DNS_IPV6_SERVER_ADDRESS = config.LIBTELIO_DNS_IPV6
 
 
 @pytest.mark.asyncio
 @pytest.mark.parametrize(
-    "dns_server_address",
+    "alpha_ip_stack",
     [
-        pytest.param(DNS_IPV4_SERVER_ADDRESS, id="IPv4"),
-        pytest.param(DNS_IPV6_SERVER_ADDRESS, id="IPv6"),
+        pytest.param(
+            IPStack.IPv4,
+            marks=pytest.mark.ipv4,
+        ),
+        pytest.param(
+            IPStack.IPv6,
+            marks=pytest.mark.ipv6,
+        ),
+        pytest.param(
+            IPStack.IPv4v6,
+            marks=pytest.mark.ipv4v6,
+        ),
+    ],
+)
+@pytest.mark.parametrize(
+    "beta_ip_stack",
+    [
+        pytest.param(
+            IPStack.IPv4,
+            marks=pytest.mark.ipv4,
+        ),
+        pytest.param(
+            IPStack.IPv6,
+            marks=pytest.mark.ipv6,
+        ),
+        pytest.param(
+            IPStack.IPv4v6,
+            marks=pytest.mark.ipv4v6,
+        ),
     ],
 )
-async def test_dns(dns_server_address: str) -> None:
+async def test_dns(
+    alpha_ip_stack: IPStack,
+    beta_ip_stack: IPStack,
+) -> None:
     async with AsyncExitStack() as exit_stack:
+        dns_server_address_alpha = ""
+        dns_server_address_beta = ""
+
+        if alpha_ip_stack in [IPStack.IPv4, IPStack.IPv4v6]:
```

Comment:
Maybe it would make sense to have a constant map `ip_stack -> DNS_IP_SERVER_ADDRESS`. It would allow to simplify a bit this DNS server address enumeration :)

Reply:
It _can_ be done, but I left it un-mapped intentionally to mix-up addresses in some tests, when there's `IPv4v6` stack used.

Reply:
In any case, we should avoid duplication here somehow, where in the tests themselves (at least ones which are not directly related to IPv6 like DNS) the tests themselves would not be enumerating stacks...

Reply:
Same as [this](https://github.com/NordSecurity/libtelio/pull/96#discussion_r1336823260)